qiskit/circuit/quantumcircuit.py

```diff
@@ -2134,33 +2141,57 @@ def cu(self, theta, phi, lam, gamma,   # pylint: disable=invalid-name
     def u1(self, theta, qubit):  # pylint: disable=invalid-name
         """Apply :class:`~qiskit.circuit.library.U1Gate`."""
         from .library.standard_gates.u1 import U1Gate
+        warnings.warn('The QuantumCircuit.u1 method is deprecated as of 0.16.0. It will be removed '
+                      'no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.p method instead, which acts identically.',
+                      DeprecationWarning, stacklevel=2)
         return self.append(U1Gate(theta), [qubit], [])
 
     def cu1(self, theta, control_qubit, target_qubit, label=None, ctrl_state=None):
         """Apply :class:`~qiskit.circuit.library.CU1Gate`."""
         from .library.standard_gates.u1 import CU1Gate
+        warnings.warn('The QuantumCircuit.cu1 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.cp method instead, which acts identically.',
+                      DeprecationWarning, stacklevel=2)
         return self.append(CU1Gate(theta, label=label, ctrl_state=ctrl_state),
                            [control_qubit, target_qubit], [])
 
     def mcu1(self, lam, control_qubits, target_qubit):
         """Apply :class:`~qiskit.circuit.library.MCU1Gate`."""
         from .library.standard_gates.u1 import MCU1Gate
+        warnings.warn('The QuantumCircuit.mcu1 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.mcp method instead, which acts identically.',
+                      DeprecationWarning, stacklevel=2)
         num_ctrl_qubits = len(control_qubits)
         return self.append(MCU1Gate(lam, num_ctrl_qubits), control_qubits[:] + [target_qubit], [])
 
     def u2(self, phi, lam, qubit):  # pylint: disable=invalid-name
         """Apply :class:`~qiskit.circuit.library.U2Gate`."""
         from .library.standard_gates.u2 import U2Gate
+        warnings.warn('The QuantumCircuit.u2 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.u method instead, where u2(a, b) = u(pi/2, a, b).',
```

Comment:
```suggestion
                      'QuantumCircuit.u method instead, where u2(φ,λ) = u(pi/2, φ, λ).',
```
and can you add the decomposition in terms of 2 Phase gates and one SX

Reply:
added in a4ccdde76